### PR TITLE
Unit testing: Fixing minor grammatical errors

### PR DESCRIPTION
**Pull request by omusavi**
Some grammatical issues that persisted after the PR #370.

### CODE REVIEW

**Review by liliankasem — APPROVED** [review]

File: automated-testing/unit-testing/readme.md [hunk 1]
```diff
@@ -45,7 +45,11 @@ but the general techniques and best practices of writing a unit test are univers
 ### Techniques
 
 These are some commonly used techniques that will help when authoring unit tests. For some examples, see the pages on
-using [abstraction and dependency injection to author a unit test](authoring_example.md), or how to do [test-driven development](tdd.md)
+using [abstraction and dependency injection to author a unit test](authoring_example.md), or how to do [test-driven development](tdd_example.md).
```
Comment (RichCzyzewski) [inline automated-testing/unit-testing/readme.md:48]: There's another instance of a broken test-driven development link around line ~100 on this page.

**Review by RichCzyzewski — CHANGES_REQUESTED** [review]
Looks good, found one more broken tdd link. Fix that then should be good to go.

File: automated-testing/unit-testing/readme.md (hunk 1 above)
Comment (omusavi) [reply automated-testing/unit-testing/readme.md:48]: fixed!

**Review by RichCzyzewski — APPROVED** [review]

File: automated-testing/unit-testing/authoring_example.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # Example: Authoring a unit test
 
-To illustrate some of the unit testing techniques, let's start with an example of some code we wish to add unit tests
-for. In this example, we have a configuration class that contains all the startup options for an app we are writing.
-Normally it reads from a `.config` file, but we are having three problems with the current implementation:
+To illustrate some of the unit testing techniques for an object oriented language, let's start with an example of some
+code we wish to add unit tests for. In this example, we have a configuration class that contains all the startup options
+for an app we are writing. Normally it reads from a `.config` file, but we are having three problems with the current
+implementation:
```
Comment (davidkpiano) [inline automated-testing/unit-testing/authoring_example.md:6]:
```suggestion
To illustrate some of the unit testing techniques for an object oriented language, let's start with an example of some code we wish to add unit tests for. In this example, we have a configuration class that contains all the startup options for an app we are writing. Normally it reads from a `.config` file, but we are having three problems with the current implementation:
```

File: automated-testing/unit-testing/mocking.md [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 One of the key components of writing unit tests is to remove the dependencies your system has and replacing it with an
-implementation you control. The most common method people as the replacement for the dependency is a mock, and mocking
-frameworks exist to help make this process easier.
-
-Many frameworks and articles use different meanings for the differences between test doubles. For this sake of clarity,
-this article will be using the [definitions provided by Martin Fowler](https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs). The most commonly
-used form of test double is Mocks, but there are many cases where Mocks perhaps are not the best choice and Fakes should
-be considered instead.
+implementation you control. The most common method people use as the replacement for the dependency is a mock, and
+mocking frameworks exist to help make this process easier.
```
Comment (davidkpiano) [inline automated-testing/unit-testing/mocking.md:5]:
```suggestion
implementation you control. The most common method people use as the replacement for the dependency is a mock, and mocking frameworks exist to help make this process easier.
```

File: automated-testing/unit-testing/mocking.md [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+
+Many frameworks and articles use different meanings for the differences between test doubles. A test double is a generic
+term for any "pretend" object used in place of a real one. This term, as well as others used in this page are the
```
Comment (davidkpiano) [inline automated-testing/unit-testing/mocking.md:8]:
```suggestion
Many frameworks and articles use different meanings for the differences between test doubles. A test double is a generic term for any "pretend" object used in place of a real one. This term, as well as others used in this page are the
```

**Comment by davidkpiano** [comment]
Just realized the formatting on this file is weird - there are newlines where there shouldn't be. Might not be related, but would be a good additional fix 🙏

**Review by davidkpiano — APPROVED** [review]

**Comment by omusavi** [comment]
@davidkpiano I have all markdown files set to have a max line length of 120 for readability in PRs (at least in some places having everything makes a really long line with a horizontal scroll bar), seems like github word wraps but I already had it all configured like this and most of my other projects don't disable line length in markdownlint! If its cool with you I will keep the newlines as-is for now, and if we want to have no set line-length keeping that in its own PR of just whitespace changes

**Pull request merged**